acvm/src/pwg/mod.rs

```diff
@@ -35,17 +30,12 @@ pub enum PartialWitnessGeneratorStatus {
     /// All opcodes have been solved.
     Solved,
 
-    /// The `PartialWitnessGenerator` has encountered a request for [oracle data][Opcode::Oracle] or a Brillig [foreign call][acir::brillig_vm::Opcode::ForeignCall].
-    ///
-    /// Both of these opcodes require information from outside of the ACVM to be inserted before restarting execution.
-    /// [`Opcode::Oracle`] and [`Opcode::Brillig`] opcodes require the return values to be inserted slightly differently.
-    /// `Oracle` opcodes expect their return values to be written directly into the witness map whereas a `Brillig` foreign call
-    /// result is inserted into the `Brillig` opcode which made the call using [`UnresolvedBrilligCall::resolve`].
-    /// (Note: this means that the updated opcode must then be passed back into the ACVM to be processed further.)
+    /// The `PartialWitnessGenerator` has encountered a request for a Brillig [foreign call][acir::brillig_vm::Opcode::ForeignCall]
+    /// to retrieve information from outside of the ACVM.
+    /// The result of the foreign call is inserted into the `Brillig` opcode which made the call using [`UnresolvedBrilligCall::resolve`].
     ///
-    /// Once this is done, the `PartialWitnessGenerator` can be restarted to solve the remaining opcodes.
-    RequiresOracleData {
-        required_oracle_data: Vec<OracleData>,
+    /// Once this is done, the `PartialWitnessGenerator` can be restarted to solve the new set of opcodes.
+    RequiresForeignCall {
         unsolved_opcodes: Vec<Opcode>,
         unresolved_brillig_calls: Vec<UnresolvedBrilligCall>,
```

Comment:
Can bikeshed this: https://github.com/noir-lang/acvm/pull/363#issuecomment-1587087301

Reply:
Not sure I prefer the shorter name, what was the issue with the name? I at least see `foreign_call_wait_info` as OK compared to `wait_info`

Reply:
> what was the issue with the name?

It's a bit redundant as we have a number of instances of `foreign_call.foreign_call_wait_info` which we can shorten to  `foreign_call.wait_info`.

I'm personally not a fan of the name `wait_info` either however as it's a bit nebulous. Why did we not choose `arguments` for this as the name+inputs can just be thought of as the arguments for the foreign call?

Reply:
Opened an issue for this here: https://github.com/noir-lang/acvm/issues/396